onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2846,22 +2846,17 @@
   if (mem_type == OrtMemTypeCPUInput)
     return OrtDevice();
   if (mem_type == OrtMemTypeCPUOutput)
-    return OrtDevice(OrtDevice::CPU, OrtDevice::MemType::HOST_ACCESSIBLE, OrtDevice::VendorIds::NVIDIA,
-                     0 /*CPU device id always be 0*/);
+    return OrtDevice(OrtDevice::GPU, OrtDevice::MemType::HOST_ACCESSIBLE, OrtDevice::VendorIds::NVIDIA,
+                     default_device_.Id());
```

Comment:
need to revert the change

Reply:
I see. Keep the change for device_id, but need to change to device type back to CPU.

Reply:
It's confusing while saying a device memory (GPU memory) is host (CPU) accessible. Because physically it's CPU memory. May makes more sense in another way like the CPU memory is device (GPU, NPU) accessible.

Reply:
agree. host accessible memory is memory cpu can access but the memory does not need to reside on cpu.